### PR TITLE
chore: Add integration tests for influxdb2_client against InfluxDB 2.0 OSS

### PR DESCRIPTION
**Pull request by carols10cents**
Ok, I think this is working now. What do you think?

### CODE REVIEW

**Review by lukebond — APPROVED** [review]
can't confidently review the rust tests but the CI setup LGTM!

File: influxdb2_client/tests/common/server_fixture.rs [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+pub struct ServerFixture {
+    server: Arc<TestServer>,
+}
+
+impl ServerFixture {
+    /// Create a new server fixture and wait for it to be ready. This
+    /// is called "create" rather than new because it is async and
+    /// waits. The shared database is configured with a writer id and
+    /// can be used immediately
+    ///
```
Comment (alamb) [inline influxdb2_client/tests/common/server_fixture.rs:59]: I think the reference to writer_id may be left over

File: influxdb2_client/tests/common/server_fixture.rs [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+                    }
+                    Err(e) => {
+                        println!("Waiting for HTTP server to be up: {}", e);
+                    }
+                }
+                interval.tick().await;
+            }
+        };
+
+        let capped_check = tokio::time::timeout(Duration::from_secs(100), try_http_connect);
```
Comment (alamb) [inline influxdb2_client/tests/common/server_fixture.rs:239]: this is nice

**Review by alamb — APPROVED** [review]
I think it looks great @carols10cents  👍 

I reviewed the setup code and tests and then checked that the tests actually ran as part of the CI job 🏅 

```

test result: ok. 21 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/debug/deps/setup-cadc01876d1f7f99

running 3 tests
test new_server_needs_onboarded ... ok
test onboarding ... ok
test create_users ... ok
```

File: influxdb2_client/tests/common/server_fixture.rs (hunk 1 above)
Comment (carols10cents) [reply influxdb2_client/tests/common/server_fixture.rs:59]: Whoops, great catch!

**Pull request merged**